src/common/utxobased/plugin/makeUtxoWalletTools.ts

```diff
@@ -70,23 +68,47 @@ export function makeUtxoWalletTools(config: WalletToolsConfig): UTXOPluginWallet
       })
     },
 
-    getAddress(args: AddressPath): string {
+    getAddress(args: AddressPath): { address: string, legacyAddress: string } {
       const purposeType = currencyFormatToPurposeType(args.format)
       return scriptPubkeyToAddress({
         scriptPubkey: fns.getScriptPubKey(args).scriptPubkey,
         network,
         addressType: getAddressTypeFromPurposeType(purposeType),
         coin
-      }).address
+      })
     },
 
     addressToScriptPubkey(address: string): string {
-      return addressToScriptPubkey({
+      const args = {
         address,
-        addressType: getAddressTypeFromKeys(config),
         network,
         coin
-      })
+      }
+      try {
```

Comment:
I thought "addressToScriptPubkey" should already try and guess the "addressType" so why are you doing all this "try-catch"?